### PR TITLE
Staging/fix handle play not authorized

### PR DESCRIPTION
**Pull request by rodrigopavezi**

### CODE REVIEW

File: src/rise-twitter.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ export default class RiseTwitter extends HTMLElement {
         this._play();
       } else {
         console.log('_handlePlay NOT IsAuthorized');
-        this.eventHandler.emitDone();
+        this.tweet.handleError();
```
Comment (jungeunyoon) [inline src/rise-twitter.js]: @rodrigopavezi As discussed this shouldn't be calling `tweet.handleError()`, thanks!

Comment (rodrigopavezi) [reply src/rise-twitter.js]: @jungeunyoon yeah, I think the bug is in another place. Checking it now. cheers

**Comment by rodrigopavezi** [comment]
@jungeunyoon please review again. cheers

File: src/rise-twitter.js [hunk 2]
```diff
@@ -120,7 +120,6 @@ export default class RiseTwitter extends HTMLElement {
 
   _playInPreview() {
     this.tweet.displayFillerTweets();
-    this.tweet.getTransition().start();
```
Comment (jungeunyoon) [inline src/rise-twitter.js]: good catch 👍

File: src/transition.js [hunk 3]
```diff
@@ -76,13 +77,11 @@ export default class Transition {
...
-    var that = this;
-
     if (this.transitionIntervalId === null) {
-      that._startTransition();
-      this.transitionIntervalId = setInterval(function() {
-        that._startTransition();
+      this.transitionIntervalId = setInterval( () => {
+        this._startTransition();
       }, this.intervalTime);
+      this._startTransition();
```
Comment (jungeunyoon) [inline src/transition.js]: Any particular reason why this was moved?

File: src/transition.js [hunk 4]
```diff
@@ -76,13 +77,11 @@ export default class Transition {
   }
 
   _startTransitionTimer() {
-    var that = this;
-
     if (this.transitionIntervalId === null) {
-      that._startTransition();
-      this.transitionIntervalId = setInterval(function() {
-        that._startTransition();
+      this.transitionIntervalId = setInterval( () => {
```
Comment (jungeunyoon) [inline src/transition.js]: The space before the opening paren is failing an linting rule.

File: src/transition.js [hunk 5]
```diff
@@ -61,10 +61,11 @@ export default class Transition {
 
     if (currentTweetIndex === this.numOfActualTweets) {
       this._finishedTransition();
```
Comment (jungeunyoon) [inline src/transition.js]: I think we should fade out the last tweet with `$(previousTweet).fadeOut(this.fadeTime);`. It's very abrupt right now. 

Also maybe set `this._setCurrentTweetIndex `to 0 here. Noticed you removed it.

File: src/transition.js [hunk 6]
```diff
@@ -133,7 +132,6 @@ export default class Transition {
     this.waitingForUpdate = false;
     this.tweets = null;
     this.currentTweetIndex = 0;
-    this.transitionIntervalId = null;
```
Comment (jungeunyoon) [inline src/transition.js]: Why was this removed? It's vital.

File: src/transition.js (hunk 3 above)
Comment (rodrigopavezi) [reply src/transition.js]: @jungeunyoon yes, when there is only one tweet it would call finalize before starting the setInterval. cheers

File: src/transition.js (hunk 6 above)
Comment (rodrigopavezi) [reply src/transition.js]: @jungeunyoon because it is done already on the      this._stopTransitionTimer();. cheers

File: src/transition.js (hunk 4 above)
Comment (rodrigopavezi) [reply src/transition.js]: @jungeunyoon good catch. cheers

File: src/transition.js (hunk 5 above)
Comment (rodrigopavezi) [reply src/transition.js]: @jungeunyoon I think it will fade out the last tweet because this.numOfActualTweets is always greater than currentTweetIndex by one. I tested and that is how it is working. Do you see something different? cheers

Comment (rodrigopavezi) [reply src/transition.js]: @jungeunyoon regarding the _setCurrentTweetIndex to 0 it will be done by the finishTransition. cheers

**Comment by rodrigopavezi** [comment]
@jungeunyoon Pause is also working now. Please review. cheers

File: package.json [hunk 7]
```diff
@@ -15,7 +15,7 @@
     "build-stage": "NODE_ENV=stage node build/build.js"
   },
   "dependencies": {
-    "common-component": "git+https://github.com/Rise-Vision/common-component.git#v1.6.0",
+    "common-component": "git+https://github.com/Rise-Vision/common-component.git#fix/local-messaging",
```
Comment (jungeunyoon) [inline package.json]: @rodrigopavezi Let's push these changes and release `1.6.1`

**Comment by rodrigopavezi** [comment]
@jungeunyoon please review. It is showing actual tweets on viewer local now. Cheers

File: test/integration/specs/load-twitter-component.js [hunk 8]
```diff
@@ -52,10 +52,25 @@ describe('Twitter Component - Integration', () => {
     commonConfig.disconnect();
   });
 
+  const sendLicensing = () => {
+    commonConfig.broadcastMessage({
+      from: 'twitter-module',
+      topic: 'licensing-update',
+      status: 'Current',
```
Comment (jungeunyoon) [inline test/integration/specs/load-twitter-component.js]: @rodrigopavezi Don't think we need `status`.

File: src/tweet.js [hunk 9]
```diff
@@ -184,8 +192,11 @@ export default class Tweet {
 
   _updateHeader(selector, userData) {
     const div = this.shadowRoot.querySelector(selector);
+    const profileImage = document.createElement('img');
+    profileImage.className = 'profile-image';
+    profileImage.setAttribute('src', userData.profile_image_url);
+    div.querySelector('.meta').appendChild(profileImage);
```
Comment (jungeunyoon) [inline src/tweet.js]: @rodrigopavezi Could we use `prepend` for this please, thanks.

File: src/rise-twitter.js [hunk 10]
```diff
@@ -113,19 +115,23 @@ export default class RiseTwitter extends HTMLElement {
...
+    this.state.setIsPaused(false);
     this.tweet.displayFillerTweets();
-    this.tweet.getTransition().start();
   }
 
   _play() {
+    this.state.setIsPaused(false);
     console.log('_play IsAuthorized');
-    if (this.tweet.getTransition()._isPaused()) {
+    if (this.tweet.getTweets() && this.tweet.getTweets().length > 0) {
```
Comment (jungeunyoon) [inline src/rise-twitter.js]: @rodrigopavezi We still need the check for paused state here. It works now but for the future when we want to keep the tweets on the dom it'll break. Perhaps set `this.state.setIsPaused(false);` on line 132 after the check.

File: src/rise-twitter.js [hunk 11]
```diff
@@ -113,19 +115,23 @@ export default class RiseTwitter extends HTMLElement {
         this._play();
       } else {
         console.log('_handlePlay NOT IsAuthorized');
+        if (this.messaging.isConnected()) {
```
Comment (jungeunyoon) [inline src/rise-twitter.js]: @rodrigopavezi As discussed, there's some risk with this type of communication. Should be much more stable once we move away from modules.

File: src/static/template/tweet-template.html [hunk 12]
```diff
@@ -1,7 +1,6 @@
 <div class="tweet-container">
   <div class="header">
     <div class="meta">
-      <img class="profile-image" src="" />
```
Comment (jungeunyoon) [inline src/static/template/tweet-template.html]: @rodrigopavezi Good catch

File: src/rise-twitter.js (hunk 10 above)
Comment (rodrigopavezi) [reply src/rise-twitter.js]: @jungeunyoon I think we just need to now if there is already tweets to start the transition. The transition will pause when widgets pauses so we don't need to check here as at this time it will be for sure paused. Cheers

**Comment by rodrigopavezi** [comment]
@jungeunyoon changed as per review. please review again. cheers
